### PR TITLE
Fix relation line following

### PR DESCRIPTION
**Pull request by makyo**

### CODE REVIEW

File: app/views/topology/relation.js [hunk 1]
```diff
@@ -463,14 +463,7 @@ YUI.add('juju-topology-relation', function(Y) {
...
       // Signify that a relation is being drawn.
-      topo.fire('addRelationStart');
-
-      // Create the dragline and position its endpoints properly.
-      context.addRelationDragStart({service: box});
-      context.mousemove.call(
-          container.one('.topology g').getDOMNode(),
-          null, context);
-      context.addRelationStart(box, context, origin);
+      topo.fire('addRelationDragStart', {service: box});
```
Comment (makyo) [inline app/views/topology/relation.js]: This was the culprit - wrong event

File: app/views/topology/relation.js [hunk 2]
```diff
@@ -725,6 +718,21 @@ YUI.add('juju-topology-relation', function(Y) {
...
       topo.fire('addRelationEnd');
+
+    },
+
+    /**
+      Clear sigils and variables used when creating a relation.
+
+      @method clearRelationSettings
+    */
+    clearRelationSettings: function() {
```
Comment (makyo) [inline app/views/topology/relation.js]: code dedup

File: test/test_environment_view.js [hunk 3]
```diff
@@ -1147,16 +1147,6 @@ with this program.  If not, see <http://www.gnu.org/licenses/>.
                .size()
                .should.equal(1);
 
-         // Ensure that mousemove was fired and the dragline moved.
```
Comment (makyo) [inline test/test_environment_view.js]: This never worked, because dragging never worked in this use case.

**Comment by jujugui** [comment]
Test FAILed.
Refer to this link for build results (access rights to CI server needed): 
http://ci.jujugui.org:8080//job/juju-gui/3488/
Test FAILed.

**Comment by makyo** [comment]
# QA

* `make devel`
* Deploy two services that can relate
* Ensure that you can drag the line into and out of the service, destroy the relation, drag after destroying, cancel the relation by clicking the background, etc. for both committed and uncommitted relations.

**Comment by jujugui** [comment]
Test FAILed.
Refer to this link for build results (access rights to CI server needed): 
http://ci.jujugui.org:8080//job/juju-gui/3489/
Test FAILed.

**Comment by makyo** [comment]
Passing: http://ci.jujugui.org:8080/job/juju-gui/3491/console

**Comment by hatched** [comment]
:+1: Good work tracking this down it looks great

QA Notes:
I dragged mysql, apache, wordpress to the canvas and after longpressing on the wordpress service icon I was able to drag the line between the other two icons with it snapping to each one as expected. I was then able to delete and re-create the relations as one would expect. I did the same after committing

QA Issue:
If I click on the dark grey border of the wordpress service icon I get a stream of errors in the console as I move the resulting relation line around

```
Uncaught TypeError: Cannot set property 'pos' of null
Y.Base.create.addRelationDrag @ relation.js:592
(anonymous function) @ oop-debug.js:378
Y.Subscriber._notify @ event-custom-base-debug.js:1304
Y.Subscriber.notify @ event-custom-base-debug.js:1333
Y.CustomEvent._notify @ event-custom-base-debug.js:992
Y.CustomEvent._procSubs @ event-custom-base-debug.js:1122
CEProto.fireComplex @ event-custom-complex-debug.js:226
Y.CustomEvent._fire @ event-custom-base-debug.js:1068
ET.fire @ event-custom-base-debug.js:2230
Y.Base.create.drag @ service.js:1129
(anonymous function) @ service.js:1200
event @ d3.js:705(anonymous function) @ d3.js:771
moved @ d3.js:1183(anonymous function) @ d3.js:861
```

**Comment by jujugui** [comment]
Test FAILed.
Refer to this link for build results (access rights to CI server needed): 
http://ci.jujugui.org:8080//job/juju-gui/3493/
Test FAILed.

**Comment by hatched** [comment]
Thanks that last commit fixed the QA issue!
QA :ok: !

**Comment by jujugui** [comment]
Test FAILed.
Refer to this link for build results (access rights to CI server needed): 
http://ci.jujugui.org:8080//job/juju-gui/3494/
Test FAILed.

**Comment by jujugui** [comment]
Test FAILed.
Refer to this link for build results (access rights to CI server needed): 
http://ci.jujugui.org:8080//job/juju-gui/3496/
Test FAILed.

**Comment by makyo** [comment]
Good run: http://ci.jujugui.org:8080/job/juju-gui/3497/ :shipit:

**Comment by jujugui** [comment]
Status: merge request accepted. Url: http://ci.jujugui.org:8080/job/juju-gui-merge

**Pull request merged**